token ""mariadb-10.1"" is not valid in preprocessor expressions

Compiling DBD-mysql-4.047 fails on debian/stretch systems:
```
cc -c  -I/opt/perl-5.24.3/lib/site_perl/5.24.3/x86_64-linux/auto/DBI -I/usr/include/mysql -DDBD_MYSQL_WITH_SSL -g  -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"4.047\" -DXS_VERSION=\"4.047\" -fPIC "-I/opt/perl-5.24.3/lib/5.24.3/x86_64-linux/CORE"   dbdimp.c
In file included from /usr/include/mysql/mysql.h:76:0,
                 from dbdimp.h:21,
                 from dbdimp.c:15:
dbdimp.c: In function 'mysql_dr_connect':
/usr/include/mysql/mysql_version.h:16:31: error: token ""mariadb-10.1"" is not valid in preprocessor expressions
 #define MARIADB_BASE_VERSION  "mariadb-10.1"
                               ^
dbdimp.c:1910:56: note: in expansion of macro 'MARIADB_BASE_VERSION'
 #if (MYSQL_VERSION_ID >= 50723) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
                                                        ^~~~~~~~~~~~~~~~~~~~
/usr/include/mysql/mysql_version.h:16:31: error: token ""mariadb-10.1"" is not valid in preprocessor expressions
 #define MARIADB_BASE_VERSION  "mariadb-10.1"
                               ^
dbdimp.c:1917:56: note: in expansion of macro 'MARIADB_BASE_VERSION'
 #if (MYSQL_VERSION_ID >= 50600) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
                                                        ^~~~~~~~~~~~~~~~~~~~
dbdimp.c: In function 'mysql_st_prepare':
dbdimp.c:3200:24: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
           bind->is_null=      (_Bool*) &(fbind->is_null);
                        ^
Makefile:355: recipe for target 'dbdimp.o' failed
make: *** [dbdimp.o] Error 1
```
DBD-mysql-4.046 compiled fine on this system.

Sample problem with DBD-mysql-4.047 on CentOS 7.

"/opt/rh/rh-perl526/root/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- mysql.bs blib/arch/auto/DBD/mysql/mysql.bs 644
gcc -c  -I/opt/git/perl5lib-el7-x86_64-526/perl5lib/lib/x86_64-linux-thread-multi/auto/DBI -I/usr/include/mysql -DDBD_MYSQL_WITH_SSL -g  -D_REENTRANT -D_GNU_SOURCE -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -fwrapv -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -g   -DVERSION=\"4.047\" -DXS_VERSION=\"4.047\" -fPIC "-I/opt/rh/rh-perl526/root/usr/lib64/perl5/CORE"   dbdimp.c
In file included from /usr/include/mysql/mysql.h:72:0,
                 from dbdimp.h:21,
                 from dbdimp.c:15:
dbdimp.c: In function ‘mysql_dr_connect’:
/usr/include/mysql/mysql_version.h:16:31: error: token ""mariadb-5.5"" is not valid in preprocessor expressions
 #define MARIADB_BASE_VERSION  "mariadb-5.5"
                               ^
dbdimp.c:1910:56: note: in expansion of macro ‘MARIADB_BASE_VERSION’
 #if (MYSQL_VERSION_ID >= 50723) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
                                                        ^
/usr/include/mysql/mysql_version.h:16:31: error: token ""mariadb-5.5"" is not valid in preprocessor expressions
 #define MARIADB_BASE_VERSION  "mariadb-5.5"
                               ^
dbdimp.c:1917:56: note: in expansion of macro ‘MARIADB_BASE_VERSION’
 #if (MYSQL_VERSION_ID >= 50600) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
                                                        ^
dbdimp.c:1936:19: warning: unused variable ‘ssl_mode’ [-Wunused-variable]
      unsigned int ssl_mode;
                   ^
dbdimp.c: In function ‘mysql_st_prepare’:
dbdimp.c:3200:24: warning: assignment from incompatible pointer type [enabled by default]
           bind->is_null=      (_Bool*) &(fbind->is_null);
                        ^
make: *** [dbdimp.o] Error 1
  CAPTTOFU/DBD-mysql-4.047.tar.gz
  /usr/bin/make -- NOT OK

--- a/dbdimp.c
+++ b/dbdimp.c
@@ -1907,20 +1907,22 @@ MYSQL *mysql_dr_connect(
                          (SvTRUE(*svp) ? "utf8" : "latin1"));
         }
 
-#if (MYSQL_VERSION_ID >= 50723) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
+#ifndef MARIADB_BASE_VERSION
+#if (MYSQL_VERSION_ID >= 50723)
         if ((svp = hv_fetch(hv, "mysql_get_server_pubkey", 23, FALSE)) && *svp && SvTRUE(*svp)) {
           my_bool server_get_pubkey = 1;
           mysql_options(sock, MYSQL_OPT_GET_SERVER_PUBLIC_KEY, &server_get_pubkey);
         }
 #endif
 
-#if (MYSQL_VERSION_ID >= 50600) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
+#if (MYSQL_VERSION_ID >= 50600)
         if ((svp = hv_fetch(hv, "mysql_server_pubkey", 19, FALSE)) && *svp) {
           STRLEN plen;
           char *server_pubkey = SvPV(*svp, plen);
           mysql_options(sock, MYSQL_SERVER_PUBLIC_KEY, server_pubkey);
         }
 #endif
+#endif /* MARIADB_BASE_VERSION */
 
 	if ((svp = hv_fetch(hv, "mysql_ssl", 9, FALSE)) && *svp && SvTRUE(*svp))
           {

